libtsuba/src/RDG.cpp

```diff
@@ -267,25 +272,165 @@ tsuba::RDG::WritePartArrays(const katana::Uri& dir, tsuba::WriteGroup* desc) {
   return next_properties;
 }
 
+katana::Result<void>
+tsuba::RDG::DoStoreTopology(
+    RDGHandle handle, std::unique_ptr<FileFrame> topology_ff,
+    std::unique_ptr<WriteGroup>& write_group) {
+  if (!topology_ff && !topology_file_storage().Valid()) {
+    return KATANA_ERROR(
+        ErrorCode::InvalidArgument,
+        "no topology file frame update, but topology_file_storage is invalid");
+  }
+
+  if (topology_ff) {
+    // we have an update, store the passed in memory state
+    katana::Uri path_uri = MakeTopologyFileName(handle);
+    topology_ff->Bind(path_uri.string());
+    TSUBA_PTP(internal::FaultSensitivity::Normal);
+    write_group->StartStore(std::move(topology_ff));
+    TSUBA_PTP(internal::FaultSensitivity::Normal);
+    core_->part_header().set_topology_path(path_uri.BaseName());
+  } else if (core_->part_header().topology_path().empty()) {
+    // we don't have an update, but we are persisting in a new location
+    // store our in memory state
+    katana::Uri path_uri = MakeTopologyFileName(handle);
+
+    TSUBA_PTP(internal::FaultSensitivity::Normal);
+    // depends on `topology_file_storage_` outliving writes
+    write_group->StartStore(
+        path_uri.string(), core_->topology_file_storage().ptr<uint8_t>(),
+        core_->topology_file_storage().size());
+    TSUBA_PTP(internal::FaultSensitivity::Normal);
+    core_->part_header().set_topology_path(path_uri.BaseName());
+  }
+  // else: no update, not persisting in a new location, so nothing for us to do
+
+  return katana::ResultSuccess();
+}
+
+katana::Result<void>
+tsuba::RDG::DoStoreNodeEntityTypeIDArray(
```

Comment:
These two functions (Node and Edge) version are pretty much doing the same thing. Can we combine them into 1 function in the future?